### DISCUSSION TITLE
Question on a Lagrange multiplier constraint kernel

### DISCUSSION THREAD

**Discussion by xueyang94** [idaholab/moose#19181] [Q&A Modules: Phase field] [answered]
Hello,

In the SwitchingFunctionConstraintEta kernel, the residual is `_lambda[_qp] * _dh[_qp] * _test[_i][_qp]`. Is this kernel assuming that one switching function h is a function of only one phase parameter eta? For example, only the simple and high order switching functions are considered, but not the SwitchingFunctionMultiPhaseMaterial where each h is a function of multiple etas? 

If one h can depend on multiple etas, then shouldn't the kernel residual be `_lambda[_qp] * (*_dh[i][j])[_qp] * _test[_i][_qp]`? (This refers back to the [multiphase models page](https://mooseframework.inl.gov/modules/phase_field/MultiPhase/WBM.html), the first weak form equation under the lagrange multiplier constraint section.)

Thanks,
XY

**Comment by GiudGiud** [comment, marked as answer]
hello

It's a good question imo.
h is a material property (and dh its derivative) which can be arbitrarily complex and depend on any number of phase parameters. _dh is the derivative with regards to only the variable of the kernel. 

Because it's a lagrange multiplier constraint, I would want to add only the term for that particular variable to the residual for that variable. If you look at https://github.com/idaholab/moose/blob/next/modules/phase_field/test/tests/MultiPhase/acmultiinterface.i this is what is being done. 
@dschwen will know more than me. 

Guillaume
